### PR TITLE
Integrate DTW calculation in training loop

### PR DESCRIPTION
**Pull request by RongLirr**

### CODE REVIEW

File: fluent_pose_synthesis/train.py [hunk 1]
```diff
@@ -1,7 +1,13 @@
 import sys
+import os
+sys.path.insert(0, os.path.abspath(os.path.dirname(__file__)))
+sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
```
Comment (AmitMY) [inline fluent_pose_synthesis/train.py]: you should not do this. importing should work without it

File: fluent_pose_synthesis/core/training.py [hunk 2]
```diff
@@ -199,6 +229,317 @@ def diffuse(
...
+            if self.logger: self.logger.warning("No poses collected during validation for DTW calculation.")
+            self.model.train()
+            return float('inf')
+
+        if self.logger: self.logger.info(f"Calculating DTW for {len(all_references_poses)} validation samples...")
+        dtw_score = float('inf')
+        try:
+            scores = [
+                self.validation_metric_calculator._pose_score(pred, ref)
+                for pred, ref in zip(all_predictions_poses, all_references_poses)
```
Comment (AmitMY) [inline fluent_pose_synthesis/core/training.py]: `self.validation_metric_calculator.corpus_score`

File: fluent_pose_synthesis/core/training.py [hunk 3]
```diff
@@ -77,13 +77,43 @@ def __init__(
...
+        # Initialize DTW metric calculator
+        default_dtw_dist_val = 10.0
+        self.validation_metric_calculator = DistanceMetric(
+            name="Validation DTW",
+            distance_measure=DTWDTAIImplementationDistanceMeasure(
+                name="dtaiDTW",
+                use_fast=True,
+                default_distance=default_dtw_dist_val,
+            ),
+            pose_preprocessors=[],
```
Comment (AmitMY) [inline fluent_pose_synthesis/core/training.py]: there is a processor called osmething like - "normalizeshoulders"

**Review by AmitMY — CHANGES_REQUESTED** [review]

File: fluent_pose_synthesis/core/training.py (hunk 3 above)
Comment (RongLirr) [reply fluent_pose_synthesis/core/training.py]: I used NormalizePosesProcessor(PoseProcessor) https://github.com/sign-language-processing/pose-evaluation/blob/e4704ba6bddd28699fd7bdf8bac350838f161b94/pose_evaluation/metrics/pose_processors.py#L54

**Comment by RongLirr** [comment]
1. Added AMP support as a configurable option.
2. Saving a fixed set of validation predictions each epoch for consistent visualization.
3. Computing DTW score and validation loss at every epoch.

**Comment by RongLirr** [comment]
Main changes:

1. In addition to the original MSE loss, velocity loss (and optionally, acceleration loss, but now it's turned off) has been incorporated for more smooth motion.

2. During training, setting cond_mask_prob > 0. During inference, CFG sampling logic is added to the validation pipeline, using a guidance_scale parameter (set to 2 now) to amplify the influence of the conditional input.
